hubbard/plot/spectrum.py

```diff
@@ -149,36 +144,116 @@ def __init__(self, HubbardHamiltonian, k=[0, 0, 0], spin=0,
-        if projection.upper() == '1D':
-            title += r': origin [%.2f,%.2f,%.2f] (\AA)' % tuple(origin)
-        self.set_title(title)
+
         self.set_xlabel(r'distance along [%.2f,%.2f,%.2f] (\AA)' % tuple(direction))
         self.set_ylabel(r'$E-E_\mathrm{midgap}$ (eV)')
         self.set_xlim(xmin, xmax)
         self.set_ylim(emin, emax)
         self.axes.set_aspect('auto')
 
 
-class DOS_distribution(GeometryPlot):
+class LDOS_from_eigenstate(GeometryPlot):
     """ Plot LDOS in the configuration space for the `hubbard.HubbardHamiltonian` object
 
     Parameters
     ----------
     HH: HubbardHamiltonian
         mean-field Hubbard Hamiltonian
-    PDOS: numpy.ndarray
-        Density of states projected in atomic sites obtained at certain energy
+    WF: numpy.ndarray or sisl.physics.electron.EigenstateElectron
+        Eigenstate to be plotted as LDOS, WF can be an array-like containing more than WF
+        The WF have to be stored along the second axis, i.e. each WF is stored as column vectors
+        in this case this method sums the EigenLDOS corresponding to each WF in the grid
+        This can be handy if one wants to plot the LDOS = LDOS_up + LDOS_dn
+    realspace:
+        If True it will plot the LDOS in a realspace grid otherwise it plots it as a scatter plot (PDOS)
+        with varying size depending on the PDOS numerical value
+        In this case the `z` kwarg needs to be passed to slice the real space grid at the desired z coordinate
+    """
+
+    def __init__(self, HH, wavefunction, sites=[], ext_geom=None, realspace=False, **kwargs):
+
+        # Set default kwargs
+        if realspace:
+            if 'facecolor' not in kwargs:
+                kwargs['facecolor'] = 'None'
+            if 'cmap' not in kwargs:
+                kwargs['cmap'] = 'Greys'
+        else:
+            if 'cmap' not in kwargs:
+                kwargs['cmap'] = plt.cm.bwr
+
+        super().__init__(HH.geometry, ext_geom=ext_geom, **kwargs)
+
+        x = HH.geometry[:, 0]
+        y = HH.geometry[:, 1]
+
+        # Ensure WF is an array of correct dimensions
+        WF = np.array(wavefunction)
+        if WF.shape == (len(WF),):
+            WF = np.expand_dims(WF, axis=1)
+        if realspace:
+            if 'shape' not in kwargs:
+                kwargs['shape'] = [100,100,1]
+            if 'z' not in kwargs:
+                raise ValueError('z coordinate needs to be passed to slice the real space grid')
+
+
+            if 'vmin' not in kwargs:
+                kwargs['vmin'] = 0
+
+            xmin, xmax, ymin, ymax = self.xmin, self.xmax, self.ymin, self.ymax
+
+            grid = 0
+            for i in range(WF.shape[1]):
+                dos = WF[:,i]
+                grid_i = real_space_grid(self.geometry, dos, kwargs['shape'], xmin, xmax, ymin, ymax, kwargs['z'], mode='wavefunction')
+                # Slice it to obtain a 2D grid
+                slice_grid = grid_i.grid[:, :, 0].T.real
```

Comment:
I believe this is incorrect?

What you want is to plot the absolute value of the wavefunction, no?
So `grid += np.absolute(grid_i.grid[:, :, 0])`. This is `(real ** 2 - imag ** 2)**0.5`, if you want it squared, then do that.

Reply:
Yes thanks!!

Reply:
thanks for correcting my sign mistake! ;)